Tests/ComposableArchitectureTests/AppStorageTests.swift

```diff
@@ -14,6 +14,16 @@ final class AppStorageTests: XCTestCase {
     XCTAssertEqual(defaults.integer(forKey: "count"), 1)
   }
 
+  func testDefaultsRegistered() {
+    @Dependency(\.defaultAppStorage) var defaults
+    @Shared(.appStorage("count")) var count = 42
+    XCTAssertEqual(defaults.integer(forKey: "count"), 42)
```

Comment:
Previously this would fail because `defaults.integer(forKey: "count")` would return `0`.